go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -370,26 +370,51 @@ func addDefaultConntrackRules(nodeName, gwBridge, gwIntf string, stopChan chan s
 	nFlows := 0
 	// table 0, packets coming from pods headed externally. Commit connections
 	// so that reverse direction goes back to the pods.
-	_, stderr, err = util.RunOVSOfctl("add-flow", gwBridge,
-		fmt.Sprintf("cookie=%s, priority=100, in_port=%s, ip, "+
-			"actions=ct(commit, zone=%d), output:%s",
-			defaultOpenFlowCookie, ofportPatch, config.Default.ConntrackZone, ofportPhys))
-	if err != nil {
-		return fmt.Errorf("failed to add openflow flow to %s, stderr: %q, "+
-			"error: %v", gwBridge, stderr, err)
+	if config.IPv4Mode {
+		_, stderr, err = util.RunOVSOfctl("add-flow", gwBridge,
+			fmt.Sprintf("cookie=%s, priority=100, in_port=%s, ip, "+
+				"actions=ct(commit, zone=%d), output:%s",
+				defaultOpenFlowCookie, ofportPatch, config.Default.ConntrackZone, ofportPhys))
+		if err != nil {
+			return fmt.Errorf("failed to add openflow flow to %s, stderr: %q, "+
+				"error: %v", gwBridge, stderr, err)
+		}
+		nFlows++
+	}
+	if config.IPv6Mode {
+		_, stderr, err = util.RunOVSOfctl("add-flow", gwBridge,
+			fmt.Sprintf("cookie=%s, priority=100, in_port=%s, ipv6, "+
+				"actions=ct(commit, zone=%d), output:%s",
+				defaultOpenFlowCookie, ofportPatch, config.Default.ConntrackZone, ofportPhys))
+		if err != nil {
+			return fmt.Errorf("failed to add openflow flow to %s, stderr: %q, "+
+				"error: %v", gwBridge, stderr, err)
+		}
+		nFlows++
 	}
-	nFlows++
 
 	// table 0, packets coming from external. Send it through conntrack and
 	// resubmit to table 1 to know the state of the connection.
-	_, stderr, err = util.RunOVSOfctl("add-flow", gwBridge,
-		fmt.Sprintf("cookie=%s, priority=50, in_port=%s, ip, "+
-			"actions=ct(zone=%d, table=1)", defaultOpenFlowCookie, ofportPhys, config.Default.ConntrackZone))
-	if err != nil {
-		return fmt.Errorf("failed to add openflow flow to %s, stderr: %q, "+
-			"error: %v", gwBridge, stderr, err)
+	if config.IPv4Mode {
+		_, stderr, err = util.RunOVSOfctl("add-flow", gwBridge,
+			fmt.Sprintf("cookie=%s, priority=50, in_port=%s, ip, "+
+				"actions=ct(zone=%d, table=1)", defaultOpenFlowCookie, ofportPhys, config.Default.ConntrackZone))
```

Comment:
might be clearer to do

    if config.IPv4Mode {
            // pod-to-external flow
            // external-to-pod flow
    }
    if config.IPv6Mode {
            // pod-to-external flow
            // external-to-pod flow
    }

rather than

    // pod-to-external flow
    if config.IPv4Mode {
            ...
    }
    if config.IPv6Mode {
            ...
    }
    // external-to-pod flow
    if config.IPv4Mode {
            ...
    }
    if config.IPv6Mode {
            ...
    }

Reply:
Done.